### PR TITLE
rename to create XMP

### LINKED ISSUE #15144: Rename "write sidecar file for each image" to "create sidecar file for each image"
### Describe the bug

.

### Steps to reproduce

discussed at https://github.com/darktable-org/darktable/issues/3632

### Expected behavior

_No response_

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you install darktable from?

darktable.org

### darktable version

master

### What OS are you using?

Mac

### What is the version of your OS?

13

### Describe your system?

_No response_

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

### PR DESCRIPTION
**Pull request by fulldecent**
fixes #15144

### CODE REVIEW

**Comment by fulldecent** [comment]
The word "create" (in computers) means that a file did not exist before and now it exists.

The word "write" (in computers) means two things, ambiguously. It means that if a file did not exist before and now it exists. And it also means that a file previously contained some contents and now it contains new contents.

For example, in terminal if you run `echo 'hi' >> file` command twice. The first time is a "create" and also it can be considered a "write". The second time it is also a "write" but it is not a "create".

---

In darktable, the setting that I am changing controls when the file is created. It does NOT control when the files contents are changed.

Therefore it is appropriate to abandon the "write" word and to adopt the "create" word.

**Comment by TurboGit** [comment]
> The word "create" (in computers) means that a file did not exist before and now it exists.

But in general language write means that the file is created or changed (write does not convey the fact that the file existed before or not) and this is exactly what is happening in darktable. If you prefer the first write does create the file, the subsequent calls are just overwriting it with new content.

**Comment by fulldecent** [comment]
Yes. The file is rewritten/modified/updated when the photo is edits.*

But that is not the purpose of this preference selection. 

This preference selected at which time the file will go from not existing to existing. 

If the preference is selected "on import" then the time when the file goes from not existing to existing is when the photos are imported. 

Otherwise is the preference is selected "on edit" then the time when the file goes from not existing to existing is the time when the photo is first edited. 

Summarizing, the time when the sidecar file goes from not existing to existing (I.e. it is created) is either a) when imported or b) when first edited. 

And the succinct way to say this is "when to create the sidecar file? (On import, or when photo edited)"

---

However the current wording (when select to the former) says "write the file when importing". This also implies DO NOT WRITE the file AFTER IMPORT. 

Therefore the current wording is incorrect. 

---




* ignoring the =NONE case

**Comment by fulldecent** [comment]
Screenshot 2023-08-31 at 16 10 41

**Comment by gi-man** [comment]
Your original issue was that you setup dt to write xmp on edits. You do not edit your images. You tagged images and expected the tags to be written into the none existing xmp. 

Rewording 'write' to 'create' does not change your original issue. If you still need to press the button or change your settings to write xmp on import.

**Comment by fulldecent** [comment]
It does fix it. 

Because if I selected the "create" option for "on import", as it should be correctly be called, then I would have a side car file. 

And then when I set the image rating it would be in the sidecar file. 

But I did not select the "write on import" option because I though it would ONLY write on import and it would NOT write on any other occasion.

**Comment by gi-man** [comment]
> write on import" option because I though it would ONLY write on import

I understand you had an incorrect assumption. Why would we only create the xmp and never update it? The manual is clear on how this works:

`on import: An XMP file will be written as soon as you add an image to darktable’s library and will be subsequently updated each time you edit it`

**Comment by fulldecent** [comment]
I had this understanding because the choices were "write on import" or "write on edit".

If those are the choices then obviously I want "write on edit" because otherwise they will get stale.

If instead the wording was corrected to "create on import" and "create on edit" then nobody will have this problem.

The UI should be usable and correct as-is without errors being reconciled through a manual.

**Comment by dterrahe** [comment]
I'm wondering what the impact might be of changing the .po files even for languages where you haven't adjusted the translation. Would the official maintainer still get a notification that their wording might be stale?

**Comment by TurboGit** [comment]
The .po files must not be touched except by the translators.

**Comment by TurboGit** [comment]
@dterrahe : BTW, as I have no strong opinion, do you also find `create` a better wording? 

@elstoc ? I'm not an English native speaker so it is very hard for me to understand this semantic change.

**Comment by elstoc** [comment]
I kind of agree that create is a better description of what it does but I don't know if it's worth breaking everyone's preferences for -- anyone who has changed this preference in the past will have it reset to default when they upgrade and it won't necessarily be obvious to them (not everyone reads the release notes)

**Comment by kmilos** [comment]
As I already mentioned, instead of semantics, why not consider behaviour change: treat lighttable modifications as edits? Or introduce another option "write on `lighttable change`"?

**Comment by gi-man** [comment]
> As I already mentioned, instead of semantics, why not consider behaviour change: treat lighttable modifications as edits? Or introduce another option "write on `lighttable change`"?

I fully agree that this might be a better approach than just semantics. I would reduce it to just two options: create/use xmp (default) OR no xmp.

**Comment by dterrahe** [comment]
I don't think this change is likely to cause any problems (except for translators) since the name of the pref and the options are unchanged. So if it even makes one person happy, why not? I like it when people don't just complain but take responsibility for improving things (although just bug reporting is appreciated too!)

For me this would have made zero difference since as with any promised software feature I follow a strict trust-but-verify policy.

**Comment by elstoc** [comment]
Yeah sorry I thought it was keyed on description. I was wrong.

**Comment by victoryforce** [comment]
> I think this change probably will make it clearer for new users.

I agree. @TurboGit My vote is to accept this change.

**Comment by fulldecent** [comment]
To confirm, I should undo all .po localization changes, right?

**Comment by TurboGit** [comment]
> To confirm, I should undo all .po localization changes, right?

Yes please.

**Comment by victoryforce** [comment]
> To confirm, I should undo all .po localization changes, right?

Yes, that's for sure.

**Comment by fulldecent** [comment]
Rebased and trimmed the unnecessary .po changes

**Comment by victoryforce** [comment]
After thinking about this setting carefully, I have a strong feeling that the phrase "for each image" in the option label adds nothing to the understanding at the very least, and perhaps it will confuse some users.

The fact is that initially there were only two options: never write sidecars or create them when importing _for each image_. Then the words "for each image" were at least correct and additionally explained to the user that during import, all images receive a sidecar.

After the appearance of "after edit", the clarification about "each image" is no longer true. A sidecar in this mode is created only for those individual images that the user edits. So I suggest removing the words about each image, as this can confuse the user and lead them to misunderstand. The option label could look like `create sidecar file` or (even better to my taste) `when to create sidecar file`.

What are your thoughts on this?

File: data/darktableconfig.xml.in [hunk 1]
```diff
@@ -252,7 +252,7 @@
       </enum>
     </type>
     <default>on import</default>
-    <shortdescription>write sidecar file for each image</shortdescription>
+    <shortdescription>create sidecar file for each image</shortdescription>
     <longdescription>the sidecar files hold information about all your development steps to allow flawless re-importing of image files.\n\ndepending on the selected mode sidecar files will be written:\n - 'never'\n - 'on import': immediately after importing the image\n - 'after edit': after any user change on the image.</longdescription>
```
[the remark this answers is not in the document]
Comment (fulldecent) [reply data/darktableconfig.xml.in]: Thank you, fixed in 0f14e2c4e81ad8dceee9a2bd2a9364345de5093c

**Comment by fulldecent** [comment]
Thank you @victoryforce, updated in 0f14e2c4e81ad8dceee9a2bd2a9364345de5093c

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**